### PR TITLE
test/e2e: add debugging guide for new framework

### PR DESCRIPTION
**Pull request by shashankram**
Adds a debugging guide to list basic workflows to enable running tests using a debugger.

### CODE REVIEW

**Review by sam-heilbron — COMMENTED** [review]
small notes, ill follow up with more thoughts later on

File: changelog/v1.17.0-beta24/e2e-dbg.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+changelog:
+  - type: NON_USER_FACING
+    description: |
+      E2e test debugging doc.
+    skipCI: false
```
[the remark this answers is not in the document]
Comment (shashankram) [reply changelog/v1.17.0-beta24/e2e-dbg.yaml]: I used my GME helper script! Do we have an equivalent Make target to auto-create a changelog file in the correct directory?

[the remark this answers is not in the document]
Comment (sam-heilbron) [reply changelog/v1.17.0-beta24/e2e-dbg.yaml]: https://github.com/solo-io/gloo/tree/main/devel/tools

File: test/kubernetes/e2e/debugging.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    "TestK8sGateway/Deployer",
+    "-test.v",
+  ],
+  "env": {
+    "SKIP_INSTALL": "true",
+  },
+}
+```
+
+Setting `SKIP_INSTALL` to `true` will skip the installation of Gloo, which is useful to debug against a pre-existing/stable environment with Gloo already installed.
```
[the remark this answers is not in the document]
Comment (shashankram) [reply test/kubernetes/e2e/debugging.md:45]: I disagree that this is the wrong place to reference. It's something I use to quickly debug tests that is worth noting.

File: test/kubernetes/e2e/debugging.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
+# Debugging E2e Tests
+
+This document describes workflows that may be useful when debugging e2e tests with an IDE's debugger.
+
+## Overview
+
+The entry point for an e2e test is a Go test function of the form `func TestXyz(t *testing.T)` which represents a top level suite against an installation mode of Gloo. For example, the `TestK8sGateway` function in `test/kubernetes/e2e/k8sgateway/k8s_gw_test.go` represents a suite of feature tests against an installation of Gloo.
```
Comment (sheidkamp) [inline test/kubernetes/e2e/debugging.md:7]: Consider adding a link into the code

File: test/kubernetes/e2e/debugging.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+Each feature suite is invoked as a subtest of the top level suite. The subtests use [testify](https://github.com/stretchr/testify) to structure the tests in the feature's test suite and make use of the libarary's assertions.
```
Comment (sheidkamp) [inline test/kubernetes/e2e/debugging.md:9]: You are referring to "Top Level Suites", "Feature Suites", and "Subtests" in this section - can you define each of these and/or provide links?

**Review by sheidkamp — CHANGES_REQUESTED** [review]
Looks good! I was able to get the tests running easily.

I got a little confused about what some terms meant, and have comments where that happened.

File: changelog/v1.17.0-beta24/e2e-dbg.yaml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+changelog:
```
Comment (sheidkamp) [inline changelog/v1.17.0-beta24/e2e-dbg.yaml:1]: Comment to allow updated review

**Review by sheidkamp — DISMISSED** [review]

File: test/kubernetes/e2e/debugging.md (hunk 4 above)
Comment (shashankram) [reply test/kubernetes/e2e/debugging.md:9]: rephrased this, but the lines above are meant to convey these terms.

**Review by sheidkamp — DISMISSED** [review]

**Review by sam-heilbron — APPROVED** [review]

**Review by sheidkamp — APPROVED** [review]

**Pull request merged**